src/sif/recost.cc

```diff
@@ -125,7 +125,7 @@ void recost_forward(baldr::GraphReader& reader,
     length += edge->length() * edge_pct;
     // construct the label
     label = EdgeLabel(predecessor++, edge_id, edge, cost, cost.cost, 0, costing.travel_mode(), length,
-                      transition_cost, time_restrictions_TODO);
+                      transition_cost, time_restrictions_TODO, true);
```

Comment:
i think this would be more correct as `false` so that you dont recost cost a multiple edge closer and fail the allowed check because pruning is turned on. maybe it might be better instead of hardcoding here to actually use `!ignore_access`

Reply:
Yeah I didn't fully grasp what recost costing was used for. So wherever it wasn't clear to me what the costing was doing, I defaulted to `true` to keep it same as before.

Can using `!ignore_access` cause issues because it will also ignore intermediate closures?

Reply:
TLDR; basically what it says is "be permissive" when recomputing the cost of the route. 

recosting is used to compute cost/time of a path using a different costing. it does call the "allowed" function using the labels it generates while recomputing the cost of a path with a different costing. this means that if you turn on pruning the allowed function will return `false` if the original path was going through a multiple edge closure. while i think it is more correct to set pruning to `false` when someone sets `ignore_access` it turns out that it wont matter because the `if` above checks if that boolean was set. so with the change you just made you'll have made it the case that `allowed` will return `true` and not try to fail the route at a closure. honestly it might be pretty silly at all to even check the allowed method in recosting. i did that in the original implementation but the more times we revisit it the less i like that it exists :smile: 

in the future though it will be important because its the way that we get information about any kinds of restrictions the path might go through which we will want to bubble up to the user.